packages/graphql-language-service-parser/src/benchmark/index.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable no-console */
+import fs from 'fs';
+import path from 'path';
+import Benchmark from 'benchmark';
+import { parse } from 'graphql';
+import { onlineParser, CharacterStream } from '../';
+
+interface IStats {
+  mean: number;
+  rme: number;
+  variance: number;
+}
+
+const printResult = (stats: IStats, name: string, schema: string) => {
+  console.log({
+    name,
+    mean: `${1.0 / stats.mean} ops / sec`,
+    variance: stats.variance,
+    rme: `${stats.rme}%`,
+    lines: schema.split('\n').length,
+  });
+
+  console.log(`Completed test suite: ${name}`);
+};
+
+const runSplitTest = (name: string, schema: string) => {
+  const stats: IStats[] = [];
+
+  const suite = new Benchmark.Suite();
+  const parser = onlineParser();
+  let state: any = parser.startState();
+
+  schema.split('\n').forEach((line, index) => {
+    let prevState: any;
+    let completeState: any;
+
+    suite.add({
+      maxTime: 0.1,
+      onStart: () => {
+        prevState = { ...state };
+      },
+      fn: () => {
+        const stream = new CharacterStream(line);
+
+        while (!stream.eol()) {
+          parser.token(stream, state);
+          if (state.kind === 'Invalid') {
+            console.log(state.kind, line, index);
+            throw new Error('Invalid');
+          }
+        }
+
+        completeState = state;
+        state = { ...prevState };
+      },
+      onError: console.log,
+      onComplete: (e: any) => {
+        state = completeState;
+        stats.push(e.target.stats);
+      },
+    });
+  });
+
+  console.log(`Started test suite: ${name}`);
+
+  suite.run();
+
+  const results = stats.reduce(
+    (result, stat) => {
+      result.mean += stat.mean / stats.length;
+      result.rme += stat.rme / stats.length;
+      result.variance += stat.variance / stats.length;
+
+      return result;
+    },
+    { mean: 0, variance: 0, rme: 0 },
+  );
+
+  printResult(results, name, schema);
+};
+
+const runWholeTest = (name: string, schema: string) => {
+  const suite = new Benchmark.Suite('', {
+    onComplete: (e: any) => printResult(e.target.stats, name, schema),
+  });
+
+  const parser = onlineParser();
+  const state: any = parser.startState();
```

Comment:
you can use `State` from `graphql-language-service` or `graphql-language-service-parser` here if you want